src/applications/appeals/shared/validations/index.js

```diff
@@ -11,3 +11,26 @@ export const validateRequireRatedDisability = (
     errors.addError?.(errorMsgs.contestedIssue);
   }
 };
+
+/**
+ * Check validations for Custom pages
+ * @param {Function[]} validations - array of validation functions
+ * @param {*} data - field data passed to the validation function
+ * @param {*} fullData - full and appStateData passed to validation function
+ * @param {*} index - array index if within an array
+ * @returns {String[]} - error messages
+ */
+export const checkValidations = (
+  validations = [],
+  data = {},
+  fullData = {},
+  index,
+) => {
+  const errors = { errorMessages: [] };
+  errors.addError = message => errors.errorMessages.push(message);
+  /* errors, fieldData, formData, schema, uiSchema, index, appStateData */
+  validations.map(validation =>
+    validation(errors, data, fullData, null, null, index, fullData),
```

Comment:
What does index represent here? Noticing it wasn't there previously.

Reply:
We're emulating the form system validation functions with the following parameters:

```
/* errors, fieldData, formData, schema, uiSchema, index, appStateData */
```

Where index is the page index within the data array. So if a page path includes an `:index`, then this same index is passed into the validation function